src/QsCompiler/TextProcessor/QsFragmentParsing.fs

```diff
@@ -59,15 +59,31 @@ let private buildSymbolTuple (items, range : Position * Position) =
 let private expectedIdentifierDeclaration continuation = 
     expected localIdentifier ErrorCode.InvalidIdentifierDeclaration ErrorCode.MissingIdentifierDeclaration invalidSymbol continuation 
 
+/// Uses multiSegmentSymbol to parse the name of a namespace, 
+/// concatenates all path segments and the symbol with a dot, and returns a the concatenated string as Some.
+/// Returns None if the path contains segments that are None (i.e. invalid).
+/// Generates a suitable diagnostic if the namespace name ends in an underscore. 
+let internal namespaceName = // internal for testing purposes
+    let asNamespaceName ((path, sym : string option), range : Position * Position) = 
+        let names = [for segment in path do yield segment ] @ [sym]
```

Comment:
Not sure if path is a list or a seq, but wouldn't this be `path @ [sym]` or `Seq.toList path @ [sym]`?